roles/pulp/tasks/main.yaml

```
@@ -0,0 +1,24 @@
+---
+- name: Pull the Pulp container image
+  containers.podman.podman_image:
+    name: "{{ pulp_image }}"
+    state: present
+
+- name: Deploy Pulp Container
+  containers.podman.podman_container:
+    name: "{{ pulp_container_name }}"
+    image: "{{ pulp_image }}"
+    state: started
+    restart_policy: always
+    ports:  
+      - "{{ pulp_ports }}"
+    volumes:
+      - "{{ pulp_volumes }}"
```

```suggestion
    ports: "{{ pulp_ports }}"
    volumes: "{{ pulp_volumes }}"
```
both, `pulp_ports` and `pulp_volumes` are already lists, so you don't need to make them into lists here, as otherwise you're passing `ports: [ ['8080:80'] ]` here and that doesn't work.